### ISSUE TITLE
Adapting Variants of NanoGPT weights to Transformer Lens

### ISSUE THREAD

**Issue by BiEchi** [adamkarvonen/chess_llm_interpretability#2]
Hi @adamkarvonen , in your `nanogpt_to_transformer_lens.ipynb`, do you need to update the `HookedTransformer` class to adapt to your model? I find your nanogpt code not exactly the same as the official code [`model.py`](https://github.com/karpathy/nanoGPT/blob/master/model.py).

**Comment by adamkarvonen** [comment]
What differences do you notice other than this line:

`logits = self.lm_head(x[:, :, :])`

?

I didn't have to adapt Hooked transformer at all. At the bottom of the
notebook, the model is successfully loaded into a Hooked transformer.

**Comment by BiEchi** [comment]
Hi @adamkarvonen, I find that TransformerLens uses techniques like `fold_layer_norm` and `center_writing_weights`, but nanoGPT doesn't have those. How did you handle those?

**Comment by adamkarvonen** [comment]
I just used the default values. If they need to be modified, you can add the arguments to `load_and_process_state_dict`. I call that function here: https://github.com/adamkarvonen/chess_llm_interpretability/blob/main/model_setup.py#L172

**Comment by BiEchi** [comment]
No worries I checked your code out. So basically `layernorm folding + LayerNormPre layer = LayerNorm layer`. So your implementation (LN Fold + LNPre) is equivalent to Nanogpt (LN).
